app/assets/javascripts/discourse/app/lib/sidebar/common/community-section/section.js

```diff
@@ -77,7 +77,8 @@ export default class CommunitySection {
 
         return filtered;
       }, [])
-      .concat(this.apiPrimaryLinks);
+      .concat(this.apiPrimaryLinks)
+      .filter((link) => link.shouldDisplay);
```

Comment:
This part 1 of the fix. We should only take into account the links that we should display.